pkg/controller/scaledobject/hpa.go

```diff
@@ -114,6 +117,16 @@ func (r *ReconcileScaledObject) updateHPAIfNeeded(logger logr.Logger, scaledObje
 		logger.Info("Updated HPA according to ScaledObject", "HPA.Namespace", foundHpa.Namespace, "HPA.Name", foundHpa.Name)
 	}
 
+	if !equality.Semantic.DeepDerivative(hpa.ObjectMeta, foundHpa.ObjectMeta) {
```

Comment:
I am not sure if comparing the whole ObjectMeta is optimal. There are several properties that are being changed by k8s in the ObjectMeta and we don't mind that they are changed (they are not affecting the HPA itself). Comparing labels should be IMHO enough.